common/src/main/java/org/apache/iceberg/common/DynConstructors.java

```diff
@@ -84,7 +78,7 @@ public <R> R invoke(Object target, Object... args) {
 
     @Override
     @SuppressWarnings("unchecked")
-    public <R> R invokeChecked(Object target, Object... args) throws Exception {
+    <R> R invokeChecked(Object target, Object... args) throws Exception {
```

Comment:
I don't think we can make this change here without going through a deprecation cycle. Please mark this as deprecated since 1.7.0 and we can remove/make it less visible in 1.8.0